### ISSUE TITLE
Consider not using Path.GetDirectoryName in ViewStartUtility

### ISSUE THREAD

**Issue by pranavkm** [aspnet/Mvc#2009]
Today given an app relative path, we use `Path.GetDirectoryName` to determine potential paths for _ViewStart.cshtml. `Path.GetDirectoryName` does a bunch of operations to figure out if the path is absolute and sanitizing which shows up in perf profiles. Additionally, it attempts to normalize the directory separator

```C#
var path = Path.GetDirectoryName("foo/bar/baz"); // foo\bar
```

Since these paths are generated from the Razor view engines, an option would be to consider simply substring-ing the passed in path rather than using Path.GetDirectoryName. Some discussion about this is available at https://github.com/aspnet/Mvc/pull/1996#discussion_r24517836

**Comment by rynowak** [comment]
This bug isn't relevant anymore, we've already decided to cache all of this data as part of the changes in #3364. After that PR is checked in, we'll profile and identify any new issues that crop up.

Closing as this is not actionable anymore.

**Comment by pranavkm** [comment]
We actually fixed this as part of #3294. Closing this as dup
